socorro/external/postgresql/postgresqlalchemymanager.py

```diff
@@ -43,91 +33,87 @@ def __init__(self, sa_url, logger, autocommit=False):
         self.metadata.bind = self.engine
         self.session = sessionmaker(bind=self.engine)()
         self.logger = logger
-
-    def setup_admin(self):
-        self.session.execute('SET check_function_bodies = false')
+        self.on_heroku = on_heroku
+
+    # decorator for heroku stuff
+    def ignore_if_on_heroku(func):
+        def ignored(f_arg, *args, **kwargs):
+            if f_arg.on_heroku:
+                return
+            else:
+                func(f_arg, *args, **kwargs)
```

Comment:
You could just
```python
if not f_arg.on_heroku:
    return func(f_arg, *args, **kwargs)
```

Note also the important `return` that was missing. Otherwise the decorator doesn't work in a "proxy manner".